test/test_xeval.cpp

```diff
@@ -126,180 +126,179 @@ namespace xt
-            EXPECT_EQ(res_rvalue(2), 3);
-        }
-
-        TEST_F(as_strided, tensor_no_data_interface_layout_unchanged)
-        {
-            auto tensor_cast = xt::cast<int>(ten);
-            EXPECT_FALSE(HAS_DATA_INTERFACE(tensor_cast));
-            EXPECT_XTENSOR(tensor_cast);
-
-            auto res_lvalue = xt::as_strided<layout_type::row_major>(tensor_cast);
-            EXPECT_LAYOUT(res_lvalue, layout_type::row_major);
-            EXPECT_TRUE(HAS_DATA_INTERFACE(res_lvalue));
-            EXPECT_XTENSOR(res_lvalue);
-            EXPECT_EQ(res_lvalue(2), 3);
-
-            auto res_rvalue = xt::as_strided<layout_type::row_major>(
-                                xt::cast<int>(ten)
-                                );
-            EXPECT_LAYOUT(res_rvalue, layout_type::row_major);
-            EXPECT_TRUE(HAS_DATA_INTERFACE(res_rvalue));
-            EXPECT_XTENSOR(res_rvalue);
-            EXPECT_EQ(res_rvalue(2), 3);
-        }
-
-        TEST_F(as_strided, array_no_data_interface_layout_changed)
-        {
-            auto array_cast = xt::cast<int>(arr);
-            EXPECT_FALSE(HAS_DATA_INTERFACE(array_cast));
-            EXPECT_XARRAY(array_cast);
-
-            auto res_lvalue = xt::as_strided<layout_type::column_major>(array_cast);
-            EXPECT_LAYOUT(res_lvalue, layout_type::column_major);
-            EXPECT_TRUE(HAS_DATA_INTERFACE(res_lvalue));
-            EXPECT_XARRAY(res_lvalue);
-            EXPECT_EQ(res_lvalue(2), 3);
-
-            auto res_rvalue = xt::as_strided<layout_type::column_major>(
-                                xt::cast<int>(arr)
-                                );
-            EXPECT_LAYOUT(res_rvalue, layout_type::column_major);
-            EXPECT_TRUE(HAS_DATA_INTERFACE(res_rvalue));
-            EXPECT_XARRAY(res_rvalue);
-            EXPECT_EQ(res_rvalue(2), 3);
-        }
-
-        TEST_F(as_strided, tensor_no_data_interface_layout_changed)
-        {
-            auto tensor_cast = xt::cast<int>(ten);
-            EXPECT_FALSE(HAS_DATA_INTERFACE(tensor_cast));
-            EXPECT_XTENSOR(tensor_cast);
-
-            auto res_lvalue = xt::as_strided<layout_type::column_major>(tensor_cast);
-            EXPECT_LAYOUT(res_lvalue, layout_type::column_major);
-            EXPECT_TRUE(HAS_DATA_INTERFACE(res_lvalue));
-            EXPECT_XTENSOR(res_lvalue);
-            EXPECT_EQ(res_lvalue(2), 3);
-
-            auto res_rvalue = xt::as_strided<layout_type::column_major>(
-                                xt::cast<int>(ten)
-                                );
-            EXPECT_LAYOUT(res_rvalue, layout_type::column_major);
-            EXPECT_TRUE(HAS_DATA_INTERFACE(res_rvalue));
-            EXPECT_XTENSOR(res_rvalue);
-            EXPECT_EQ(res_rvalue(2), 3);
+            xt::xtensor<double, 1, layout_type::row_major> ten {1., 2., 3.2};
+            xt::xarray<double, layout_type::row_major> arr {1., 2., 3.2};
+
+            SUBCASE("array_reference")
```

Comment:
@JohanMabille  in the world of doctest  it is fine to be a subcase subcases. Thats the doctest parttern to avoid fixtures.
I could rewrite them again using classic fixtures (https://github.com/onqtam/doctest/blob/master/doc/markdown/testcases.md) but my personal preference is the subcase style.